receiver/googlecloudspannerreceiver/internal/statsreader/projectreader_test.go

```diff
@@ -20,35 +20,27 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap/zaptest"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/metadata"
 )
 
 type mockCompositeReader struct {
-	throwErrorOnRead bool
+	mock.Mock
 }
 
-func newMockCompositeReader(throwErrorOnRead bool) mockCompositeReader {
-	return mockCompositeReader{
-		throwErrorOnRead: throwErrorOnRead,
-	}
-}
-
-func (mcr mockCompositeReader) Name() string {
+func (r *mockCompositeReader) Name() string {
 	return "mockCompositeReader"
 }
 
-func (mcr mockCompositeReader) Read(context.Context) ([]*metadata.MetricsDataPoint, error) {
-	if mcr.throwErrorOnRead {
-		return nil, errors.New("error")
-	}
-
-	return []*metadata.MetricsDataPoint{{}}, nil
+func (r *mockCompositeReader) Read(ctx context.Context) ([]*metadata.MetricsDataPoint, error) {
+	args := r.Called(ctx)
+	return args.Get(0).([]*metadata.MetricsDataPoint), args.Error(1)
 }
 
-func (mcr mockCompositeReader) Shutdown() {
+func (r *mockCompositeReader) Shutdown() {
```

Comment:
```suggestion
func (mockCompositeReader) Shutdown() {
```

Reply:
This will break interface `Reader` implementation. Also mock requires pointers and I am using pointers in tests.

Reply:
Not really, once you add the method to the object the pointer also gets it. See https://play.golang.org/p/R22w3DId14z

Reply:
Ok, I tried to do as you want, and have following warning in IDE: **'Shutdown' passes a lock by the value: type 'mockCompositeReader' contains 'mock.Mock' contains 'sync.Mutex' which is 'sync.Locker'**